c2cgeoportal/lib/filter_capabilities.py

```diff
@@ -156,16 +157,19 @@ def filter_capabilities(content, role_id, wms, wms_url, headers, proxies):
     if proxies:  # pragma: no cover
         enable_proxies(proxies)
 
-    wms_structure = _wms_structure(wms_url, headers.get("Host"))
-    tmp_private_layers = list(get_private_layers())
-    for name in get_protected_layers(role_id):
-        tmp_private_layers.remove(name)
+    wms_structure_ = wms_structure(wms_url, headers.get("Host"))
+
+    gmf_private_layers = get_private_layers(wms_url)
+    for id_ in get_protected_layers(role_id, wms_url).keys():
+        if id_ in gmf_private_layers:
+            del gmf_private_layers[id_]
 
     private_layers = set()
-    for layer in tmp_private_layers:
-        private_layers.add(layer)
-        if layer in wms_structure:
-            private_layers.update(wms_structure[layer])
+    for gmflayer in gmf_private_layers.values():
+        for ogclayer in gmflayer.layer.split(","):
```

Comment:
Is the split secure ? what is in the gmflayer.layer ? the name of the layer ? Because that can be "foo, bar" no ? (Maybe not in the server side)